### PR TITLE
Fix the gpu inference for object detection

### PR DESCRIPTION
**Pull request by zhiqiangdon**
Fix the gpu inference for object detection.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/models/mmdet_image.py [hunk 1]
```diff
@@ -121,6 +123,15 @@ def forward(
         data = batch[self.image_key]
         data["img_metas"] = [img_metas.data[0] for img_metas in data["img_metas"]]
         data["img"] = [img.data[0] for img in data["img"]]
+
+        device = next(self.model.parameters()).device  # model device
```
Comment (bryanyzhu) [inline multimodal/src/autogluon/multimodal/models/mmdet_image.py:127]: Will this interfere with pytorch-lightening? Also, does this work with DDP on multi-cards or multi-nodes?

**Review by bryanyzhu — COMMENTED** [review]
Thanks for the fix.

**Review by FANGAreNotGnu — APPROVED** [review]
LGTM, thanks!

File: multimodal/src/autogluon/multimodal/models/mmdet_image.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/models/mmdet_image.py:127]: I guess it should work with ddp. Each gpu runs this code and gets its device id.

**Review by bryanyzhu — APPROVED** [review]
LGTM, thank you.

**Pull request merged**